driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -490,12 +523,17 @@ public void onSet(Connection connection, Message.Response response, long latency
                                 connection.release();
                                 assert err.infos instanceof WriteTimeoutException;
                                 WriteTimeoutException wte = (WriteTimeoutException) err.infos;
-                                retry = retryPolicy.onWriteTimeout(statement,
-                                        wte.getConsistencyLevel(),
-                                        wte.getWriteType(),
-                                        wte.getRequiredAcknowledgements(),
-                                        wte.getReceivedAcknowledgements(),
-                                        retriesByPolicy);
+                                if (statement.isIdempotentWithDefault(manager.cluster.getConfiguration().getQueryOptions()))
```

Comment:
Shouldn't we get rid of `IdempotenceAwareRetryPolicy` since we now consider idempotence outside of Retry policy?